File: atcoder.jp/contests/abs/tasks/abc087_b/main.cpp
#define _GLIBCXX_DEBUG
#include <bits/stdc++.h>
using namespace std;

template <typename T>
void print(T value) {
  cout << value << endl;
}

vector<int> range(int stop) {
  vector<int> ret(stop);
  iota(ret.begin(), ret.end(), 0);
  return ret;
}

int main() {
  int A, B, C, X;
  cin >> A >> B >> C >> X;

  int ans = 0;
  for (auto &&a : range(A + 1))
    for (auto &&b : range(B + 1))
      for (auto &&c : range(C + 1))
        if (500 * a + 100 * b + 50 * c == X) ans++;

  print(ans);
}

File: atcoder.jp/contests/abs/tasks/abc087_b/main.hs
main :: IO ()
main = do
  a <- getInt
  b <- getInt
  c <- getInt
  x <- getInt

  let ans = length [0 | i <- [0 .. a], j <- [0 .. b], k <- [0 .. c], 500 * i + 100 * j + 50 * k == x]
  print ans

getInt :: IO Int
getInt = readLn
